qutebrowser/completion/completer.py

```diff
@@ -197,13 +197,19 @@ def schedule_completion_update(self):
         For performance reasons we don't want to block here, instead we do this
         in the background.
         """
-        if (self._cmd.cursorPosition() == self._last_cursor_pos and
+        cmd = self._cmd.text().split(' ', 1)
+        if (len(cmd) > 1 and
+                0 < len(cmd[1]) < config.val.completion.min_chars and
+                self._cmd.cursorPosition() > self._last_cursor_pos):
+            log.completion.debug("Ignoring update because the length of "
+                                 "the text is less than completion.min_chars.")
+        elif (self._cmd.cursorPosition() == self._last_cursor_pos and
                 self._cmd.text() == self._last_text):
             log.completion.debug("Ignoring update because there were no "
                                  "changes.")
         else:
             log.completion.debug("Scheduling completion update.")
-            self._timer.start()
+            self._timer.start(config.val.completion.delay if self._last_text else 0)
```

Comment:
Why check for `self._last_text` here?

Reply:
because we don't want to delay the initial rendering of the completion list